### PR TITLE
add main module for ovs forwarder

### PR DESCRIPTION
**Pull request by pperiyasamy**
This is the main forwarder module hosting grpc endpoint for ovs forwarder which establishes vWire connection between service clients and endpoints.

Signed-off-by: Periyasamy Palanisamy <periyasamy.palanisamy@est.tech>

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -14,7 +14,339 @@
...
+	PCIDriversPath      string        `default:"/sys/bus/pci/drivers" desc:"path to the PCI drivers directory" split_words:"true"`
+	CgroupPath          string        `default:"/host/sys/fs/cgroup/devices" desc:"path to the host cgroup directory" split_words:"true"`
+	VFIOPath            string        `default:"/host/dev/vfio" desc:"path to the host VFIO directory" split_words:"true"`
+}
+
 func main() {
+	// ********************************************************************************
+	// setup context to catch signals
+	// ********************************************************************************
+	ctx, cancel := getTermSigalContext()
```
Comment (edwarnicke) [inline main.go:88]: I'd recommend using [signal.NotifyContext](https://github.com/networkservicemesh/cmd-forwarder-vpp/blob/e9103cf238a4b5c8449ef04170c4ccf78cbce523/main.go#L70-L80)

File: main.go [hunk 2]
```diff
@@ -14,7 +14,339 @@
...
+func getTermSigalContext() (ctx context.Context, stop context.CancelFunc) {
+	return signal.NotifyContext(
+		context.Background(),
+		os.Interrupt,
+		// More Linux signals here
+		syscall.SIGHUP,
+		syscall.SIGTERM,
+		syscall.SIGQUIT,
+	)
+}
```
Comment (edwarnicke) [inline main.go:167]: I'd counsel moving this to being inline, as it makes it clearer what's happening.  Wrapping a single function call in a function call reduces the clarity of what is going on.

Please note, this point is a matter of taste, so feel free to disagree with me here.  I offer this as advice, not as 'You need to fix this to get merged'.

File: main.go [hunk 3]
```diff
@@ -14,7 +14,339 @@
...
+	// setup context to catch signals
+	// ********************************************************************************
+	ctx, cancel := getTermSigalContext()
+	defer cancel()
+
+	setupLogger(ctx)
+
+	starttime := time.Now()
+
+	log.FromContext(ctx).Infof("executing phase 1: get config from environment (time since start: %s)", time.Since(starttime))
```
Comment (edwarnicke) [inline main.go:95]: Once again, as a matter of personal taste, I find this style: https://github.com/networkservicemesh/cmd-forwarder-vpp/blob/e9103cf238a4b5c8449ef04170c4ccf78cbce523/main.go#L111-L113 easier to read... you are more than welcome to have a differing opinion :)

File: main.go (hunk 2 above)
Comment (pperiyasamy) [reply main.go]: @edwarnicke agree with your point, have changed back using `signal.NotifyContext `as inline function call.

File: main.go (hunk 3 above)
Comment (pperiyasamy) [reply main.go:105]: sure, fixed it accordingly without making changes to linter settings. hope its ok now.

**Review by Bolodya1997 — APPROVED** [review]

**Pull request merged**